Makefile

```diff
@@ -1743,9 +1743,9 @@ endif
 		echo snappy-$(SNAPPY_VER).tar.gz checksum mismatch, expected=\"$(SNAPPY_SHA256)\" actual=\"$$SNAPPY_SHA256_ACTUAL\"; \
 		exit 1; \
 	fi
-	tar xvzf $(SNAPPY_VER).tar.gz
+	tar xvzf snappy-$(SNAPPY_VER).tar.gz
 	mkdir snappy-$(SNAPPY_VER)/build
```

Comment:
In addition to `-DPOSITION_INDEPENDENT_CODE=ON` I also had to remove this line, and make snappy build "in source" (instead of out-of-source in "build" directory), for snappy to build and link correctly. 

I get these errors with `ROCKSDB_NO_FBCODE=1 make rocksdbjavastatic -j100` if I don't do so:
```
  CC       jls/db/write_thread.o
In file included from ./util/compression.h:23:0,
                 from ./options/cf_options.h:14,
                 from ./table/format.h:25,
                 from ./table/internal_iterator.h:13,
                 from ./db/pinned_iterators_manager.h:12,
                 from ./db/range_tombstone_fragmenter.h:15,
                 from ./db/memtable.h:19,
                 from ./db/memtable_list.h:17,
                 from ./db/column_family.h:17,
                 from ./db/compaction_job.h:20,
                 from db/compaction_job.cc:10:
./snappy-1.1.7/snappy.h:45:33: fatal error: snappy-stubs-public.h: No such file or directory
 #include "snappy-stubs-public.h"
                                 ^
compilation terminated.
```
Note that `make rocksdbjavastatic` works well. It is only when I build with `ROCKSDB_NO_FBCODE=1` (which is something that we use internally to *not* depend on other FB libraries, and to just depend on vanilla open-source depedencies) I see the above erorrs ... and hence I had to get rid of building in "build" directory.

Reply:
Oh my... `-j100` what sort of laptop from the future do you have? p.s. can I have one too please?